src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -584,6 +394,22 @@ private ResponseEntity<byte[]> buildFileResponse(String path, String filename) {
         }
     }
 
+    /**
+     * Checks if the user is authorized to access an attachment
+     *
+     * @param course     the course to check if the user is part of it
+     * @param attachment the attachment for which the authentication should be checked
+     * @return true if the user is authorized to access the attachment, otherwise false is returned
+     */
+    private boolean checkAttachmentAuthorization(Course course, Attachment attachment) {
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, null);
+        if (!attachment.isVisibleToStudents() && !authCheckService.isAtLeastTeachingAssistantInCourse(course, null)) {
+            log.info("User not authorized to access attachment");
+            return false;
+        }
+        return true;
```

Comment:
```suggestion
        if (attachment.isVisibleToStudents() || authCheckService.isAtLeastTeachingAssistantInCourse(course, null)) {
            return true;
        }
        log.info("User not authorized to access attachment");
        return false;
```

The invalid value should be the fallback value (here false).

Reply:
Done in [9b9c871](https://github.com/ls1intum/Artemis/pull/5723/commits/9b9c871dbcb44fb088e2c1e5b07fefeb74a66ec8)!